src/fluree/db/json_ld/transact.cljc

```diff
@@ -535,25 +535,32 @@
    (go-try
      (let [{tx :subject did :did} (or (<? (cred/verify json-ld))
                                       {:subject json-ld})
-           opts*    (cond-> opts did (assoc :did did))
-           db*      (if-let [policy-opts (perm/policy-opts opts*)]
-                      (<? (perm/wrap-policy db policy-opts))
-                      db)
-           tx-state (->tx-state db* opts*)
-           flakes   (if (q-parse/update? tx)
-                      (<? (modify db fuel-tracker tx tx-state))
-                      (<? (insert db fuel-tracker tx tx-state)))]
-       (log/trace "stage flakes:" flakes)
-       (<? (flakes->final-db tx-state flakes))))))
+           opts*         (cond-> opts did (assoc :did did))
+           db*           (if-let [policy-opts (perm/policy-opts opts*)]
+                           (<? (perm/wrap-policy db policy-opts))
+                           db)
+           tx-state      (->tx-state db* opts*)
+           flakes-ch     (if (q-parse/update? tx)
+                           (modify db fuel-tracker tx tx-state)
+                           (insert db fuel-tracker tx tx-state))
+           fuel-error-ch (:error-ch fuel-tracker)
+           flakes-chans  (remove nil? [fuel-error-ch flakes-ch])
+           [flakes]      (alts! flakes-chans :priority true)
+           _             (when (util/exception? flakes)
+                           (throw flakes))
+           _             (log/trace "stage flakes:" flakes)
+           final-chans   (remove nil? [fuel-error-ch
```

Comment:
I'm not sure I understand what's going on here - why do we still need the fuel-error-ch? Does flakes->final-db invoke the fuel tracker some more?